src/client/core/utils/style.js

```diff
@@ -40,10 +41,14 @@ var getAncestorsAndSelf = function (node) {
     return getAncestors(node).concat([node]);
 };
 
-var isVisibilityHiddenTextNode = function (textNode) {
-    var el = domUtils.isTextNode(textNode) ? textNode.parentNode : null;
+var isVisibilityHiddenNode = function (node) {
+    var ancestors = getAncestorsAndSelf(node);
+
+    return some(ancestors, ancestor => domUtils.isElementNode(ancestor) && get(ancestor, 'visibility') === 'hidden');
+};
 
-    return el && get(el, 'visibility') === 'hidden';
+var isVisibilityHiddenTextNode = function (textNode) {
```

Comment:
we already have had `styleUtils.isNotVisibleNode` please check methods. Can we merge them?

Reply:
For my purposes I use `isElementVisible` from hammerhead.js which fits to me. As for other methods from style.js, I've just noticed that `isVisibilityHiddenTextNode` does not check all ancestors, it looks strange. So I've added `isVisibilityHiddenNode` method which checks only visibility:hidden property and then I used this method in `isVisibilityHiddenTextNode`.
As for `styleUtils.isNotVisibleNode` it also looks really strange:
```JS
export function isNotVisibleNode (node) {
    return !domUtils.isRenderedNode(node) || isHiddenNode(node) || isVisibilityHiddenTextNode(node);
}
```
It checks `visibility:hidden` only if node is text. 
I think we need to review all these methods and try to refactor them. But I would prefer to creafe the separate issue.

Reply:
It seems like `isNotVisibleNode` method has incorrect name, because it used only for text nodes
```js
if (domUtils.isTextNode(el))
        return !styleUtils.isNotVisibleNode(el);
```
But the name doesn't contains info about that.

Reply:
> I think we need to review all these methods and try to refactor them. But I would prefer to creafe the separate issue.

I agree with this, but please rename `isNotVisibleNode` -> `isNotVisibleTextNode` now. Don't you mind?